src/jit/importer.cpp

```diff
@@ -12728,6 +12728,10 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 
                     /* get a temporary for the new object */
                     lclNum = lvaGrabTemp(true DEBUGARG("NewObj constructor temp"));
+                    if (compDonotInline())
+                    {
+                        return;
```

Comment:
This is how things are done when we back out of an inline late (while importing the inlinee). We try and bail out as soon as we can.

The "too many locals logic" in `lvaGrabTemp` is causing the inline to fail, so we can't check for failure it before the call, since the inline hasn't failed at that point.